fs/vfs/fs_dup.c

```diff
@@ -65,29 +65,30 @@ int file_dup(FAR struct file *filep, int minfd, int flags)
   uint8_t f_tag_fdcheck; /* File owner fdcheck tag, init to 0 */
 #endif
 
-  fd2 = file_allocate(g_root_inode, 0, 0, NULL, minfd, true);
+  fd2 = file_allocate(g_root_inode, flags, 0, NULL, minfd, true);
   if (fd2 < 0)
     {
       return fd2;
     }
 
   ret = fs_getfilep(fd2, &filep2);
 #ifdef CONFIG_FDSAN
-  f_tag_fdsan = filep2->f_tag_fdsan;
+  f_tag_fdsan = nx_fcntl(fd2, FIOC_GETTAG_FDSAN, 0);
 #endif
 
 #ifdef CONFIG_FDCHECK
-  f_tag_fdcheck = filep2->f_tag_fdcheck;
+  f_tag_fdcheck = nx_fcntl(fd2, FIOC_GETTAG_FDCHECK, 0);
 #endif
   DEBUGASSERT(ret >= 0);
 
-  ret = file_dup3(filep, filep2, flags);
+  ret = file_dup2(filep, filep2);
```

Comment:
This is still wrong, fd2 should point to filep, not create a new file for fd2; file_allocate allocates descriptor number AND new file.

Reply:
Those are fixed now as well